### PR TITLE
feat: make Vite configuration available from distribution bundle

### PR DESCRIPTION
**Pull request by onlywei**
Addresses #155 

Adds a new config option that needs a vite plugin to work.

By default, `@fastify/vite` looks up the Vite configuration options by importing Vite from `node_modules` and then using its Node API to look up the vite.config file. This is often not desirable in production mode since many production builds declare Vite as a `devDependency` and exclude it from their final container/docker images to save space.

To support this kind of production build, this PR makes `@fastify/vite` ship with a Vite plugin that can save the handful of properties that it needs from the resolved Vite configuration object to a file in a cache directory. These properties can than be loaded _instead_ of loading the config through vite itself.

See the changes to the docs file for exactly what is happening in this change.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by galvez** [comment]
This is absolutely amazing, thank you @onlywei! I'll be reviewing shortly.

File: docs/config/index.md [hunk 1]
```diff
@@ -14,17 +14,58 @@ Enables Vite's development server if set to `true`.
 
 Its default value is set to `process.argv.includes('--dev')`, so it will automatically recognize passing the `--dev` flag to your Node.js process unless you change it to something else.
 
+### `viteConfigDistDir`
```
Comment (galvez) [inline docs/config/index.md:17]: Wasn't this renamed to `distDir`?

Comment (galvez) [reply docs/config/index.md:17]: Ah, I see now...

**Comment by onlywei** [comment]
@galvez In this PR I have made effort to maintain backwards compatibility. Shall I delete those backwards compatibility code since there is an upcoming major release anyway?

**Comment by gurgunday** [comment]
> Shall I delete those backwards compatibility code since there is an upcoming major release anyway?

Not sure if it's "too late" or not, so it could be better to keep it until @galvez chimes in

**Comment by galvez** [comment]
@mcollina Do you see foresee any issues using `node_modules/.cache`?

**Comment by galvez** [comment]
Can you explain why for instance in TypeScript's case we need `distDir` to be explicit?

```js
viteFastify({ distDir: resolve(import.meta.dirname, 'dist') }),
```

Perhaps we should include the explanation as part of the already extended bits in the docs you added?

File: packages/fastify-vite/types/index.d.ts [hunk 2]
```diff
@@ -116,6 +116,7 @@ declare namespace fastifyVite {
     renderer?: string | Partial<RendererOption>
     vite?: UserConfig
     viteConfig?: string
+    vitePluginDistDir?: string
```
Comment (galvez) [inline packages/fastify-vite/types/index.d.ts:119]: I don't see this reference anywhere else in the PR.

Comment (onlywei) [reply packages/fastify-vite/types/index.d.ts]: Forgot to delete this. Good catch. It's deleted now.

**Comment by onlywei** [comment]
> Can you explain why for instance in TypeScript's case we need `distDir` to be explicit?
> 
> ```js
> viteFastify({ distDir: resolve(import.meta.dirname, 'dist') }),
> ```
> 
> Perhaps we should include the explanation as part of the already extended bits in the docs you added?
> 
> Though it seems like something we could autoresolve?

I forgot to delete that too. Good catch. It's deleted now.

**Pull request merged**